### PR TITLE
fix: blob api doc is confusing

### PR DESCRIPTION
**Pull request by dmitry-fa**
Changes to the Javadoc of the Blob and BlobInfo classes.

### CODE REVIEW

**Review by pmakani — APPROVED** [review]

File: google-cloud-storage/src/main/java/com/google/cloud/storage/BlobInfo.java [hunk 1]
```diff
@@ -44,7 +44,21 @@
...
+ * API includes the {@link Blob} class which extends {@code BlobInfo} and declares methods to
+ * perform operations on the object. Neither {@code BlobInfo} nor {@code Blob} instances keep the
+ * object content, just the object properties.
+ *
+ * <p>Example of usage {@code BlobInfo} to crate a Storage object:
+ *
+ * <pre>{@code
+ * BlobId blobId = BlobId.of(bucketName, blobName);
+ * BlobInfo blobInfo = BlobInfo.newBuilder(blobId).setContentType("text/plain").build();
+ * Blob blob = storage.create(blobInfo, "Hello, world".getBytes(StandardCharsets.UTF_8));
```
Comment (rahulKQL) [inline google-cloud-storage/src/main/java/com/google/cloud/storage/BlobInfo.java:60]: Can you please help me to understand this line: 

> Neither {@code BlobInfo} nor {@code Blob} instances keep the object content, just the object properties.

In this block, we are creating a string object of plain text type "Hello world". The `blob` object at line 60 refers to the byte[] of "Hello world". Do you think 'blob' keeping the said object here?

From what I conclude, the quoted statement and this example are contradicting.

**Review by rahulKQL — APPROVED** [review]
Just that small doubt, beside that really nice documentation.

File: google-cloud-storage/src/main/java/com/google/cloud/storage/BlobInfo.java (hunk 1 above)
Comment (dmitry-fa) [reply google-cloud-storage/src/main/java/com/google/cloud/storage/BlobInfo.java:60]: Thanks for looking!
"Hello world" - is the content of the new created blob, it's not stored withe the 'blob' object, it is located in the Storage. So, blob.getSize() will not cause RPC request, when blob.getContent() will.

Comment (dmitry-fa) [reply google-cloud-storage/src/main/java/com/google/cloud/storage/BlobInfo.java:60]: I will provide another example for the Blob class

Comment (rahulKQL) [reply google-cloud-storage/src/main/java/com/google/cloud/storage/BlobInfo.java:60]: Thanks for clarifying my doubt!!

**Pull request closed without merging**